### ISSUE TITLE
Rule `references.keywords` should also apply to `CTE_table_name`

### ISSUE THREAD

**Issue by cmcnicoll** [sqlfluff/sqlfluff#5487]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

This should trigger RF04 for line 1.
```
with sum as (select 1 as test_rf04)

select sum.test_rf04
from sum
```

### Use case

Rule consistency for queries that use common table expressions.

### Dialect

`dialect = redshift`

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
